### PR TITLE
Emulator Tests and Cleanup

### PR DESCRIPTION
**Pull request by rakuy0**
* Standalone emulator tests
* Couple emulator fixes that might be duplicated in #394 
* Docstring fixes and snap() function
* Deal with annoying test output and a deprecation warning (#397, thanks for spotting that @williballenthin)

### CODE REVIEW

File: vivisect/analysis/generic/pointers.py [hunk 1]
```diff
@@ -72,12 +72,12 @@ def analyze(vw):
             try:
                 pname = vw.getName(ptr)
                 if pname is not None:
-                    logger.info('skipping renaming of ptr 0x%x (currently: %r)', ptr, pname)
+                    logger.debug('skipping renaming of ptr 0x%x (currently: %r)', ptr, pname)
```
Comment (atlas0fd00m) [inline vivisect/analysis/generic/pointers.py:75]: ok, now i'm confused... you want *more* logging?
or was this an accidental copy-pasta error?

File: envi/__init__.py [hunk 2]
```diff
@@ -600,16 +601,28 @@ def setEmuSnap(self, snap):
...
         self.setMemorySnap(mem)
 
+    @contextlib.contextmanager
+    def snap(self):
+        '''
+        Utility function to try something with an emulator, and then revert it.
+        '''
+        snap = self.getEmuSnap()
+        yield snap
+        self.setEmuSnap(snap)
```
Comment (williballenthin) [inline envi/__init__.py:611]: may want to do `try...finally` here or document what happens in the case of exceptions

Comment (rakuy0) [reply envi/__init__.py]: :+1:

**Review by atlas0fd00m — DISMISSED** [review]
very nice.  thanks @rakuy0 !

**Pull request merged**